config/printer-sovol-sv06-2022.cfg

```diff
@@ -0,0 +1,172 @@
+# This file contains pin mappings for the stock Sovol SV06
+# To use this config, during "make menuconfig" select the 
+# STM32F103 with a "28KiB bootloader" and serial (on USART1 PA10/PA9) communication.
+# Also, since it is using the GD32F103, please select Disable SWD at startup
+
+# Flash this firmware by copying "out/klipper.bin" to a SD card and
+# turning on the printer with the card inserted. The firmware
+# filename must end in ".bin" and must not match the last filename
+# that was flashed.
+
+# See docs/Config_Reference.md for a description of parameters.
+[include mainsail.cfg]
+
+[mcu]
+serial: /dev/serial/by-id/usb-1a86_USB2.0-Serial-if00-port0
+restart_method: command
+
+[printer]
+kinematics: cartesian
+max_velocity: 300
+max_accel: 3000
+max_z_velocity: 5
+max_z_accel: 100
+
+[stepper_x]
+step_pin: PC2
+dir_pin: !PB9
+enable_pin: !PC3
+microsteps: 16
+rotation_distance: 40
+endstop_pin: tmc2209_stepper_x:virtual_endstop
+position_endstop: 0
+position_max: 225
+homing_speed: 40
+homing_retract_dist: 0
+
+[tmc2209 stepper_x]
+uart_pin: PC1
+run_current: 0.860
+stealthchop_threshold: 180
+interpolate: False
+sense_resistor: 0.150
```

Comment:
You dont need to define interpolate and sense_resistor for every motor, remove it from all of their TMC2209 sections

Reply:
If the sense_resistor is not the default (of 0.110 for tmc2209) then it must be specified for every tmc2209 driver that does not use that default.  Similarly, if it is desired to disable interpolate then it must be disabled for every tmc2209 driver that should disable it.

@Fisheiyy - note that "printer" config files are different from "generic" config files.  For "printer" config files it is desirable to have a config that is specific to the given printer.  (For "generic" config files, we don't want the config to specify a lot of options that describe a particular printer, as there is no way to know what printer the control board will be located in).

-Kevin

Reply:
I understand but I kinda doubt the sv06 really needs a 0.04 difference in sense resistance and interpolation disabled disabled every motor, but I could be wrong, nevertheless shouldn't it be more of a base for the printer with only the basics rather than a fully configured config? Hence the removal of PA and IS

Reply:
The `sense_resistor` setting is a description of the hardware (specifically the resistance of a resistor on the control board).  Being precise is very important and some manufacturers do use values other than 0.110 ohms.

You are correct that pressure_advance should not be specified in an example config (as that describes the filament and not the printer - https://www.klipper3d.org/Example_Configs.html ).  I have no particular preference on input_shaper.

Separate from the above, the example config here would need to have a signed-off-by line.  https://www.klipper3d.org/CONTRIBUTING.html

-Kevin

Reply:
signed-off-by line added